routing/index_graph.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "routing/edge_estimator.hpp"
+#include "routing/geometry.hpp"
+#include "routing/joint.hpp"
+#include "routing/joint_index.hpp"
+#include "routing/road_index.hpp"
+#include "routing/road_point.hpp"
+
+#include "coding/reader.hpp"
+#include "coding/write_to_sink.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "std/cstdint.hpp"
+#include "std/shared_ptr.hpp"
+#include "std/unique_ptr.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+class JointEdge final
+{
+public:
+  JointEdge(Joint::Id target, double weight) : m_target(target), m_weight(weight) {}
+  Joint::Id GetTarget() const { return m_target; }
+  double GetWeight() const { return m_weight; }
+
+private:
+  // Target is vertex going to for outgoing edges, vertex going from for ingoing edges.
+  Joint::Id const m_target;
+  double const m_weight;
+};
+
+class IndexGraph final
+{
+public:
+  IndexGraph() = default;
+  explicit IndexGraph(unique_ptr<GeometryLoader> loader, shared_ptr<EdgeEstimator> estimator);
+
+  // Creates edge for points in same feature.
+  void GetDirectedEdge(uint32_t featureId, uint32_t pointFrom, uint32_t pointTo, Joint::Id target,
+                       bool forward, vector<JointEdge> & edges) const;
+  void GetNeighboringEdges(RoadPoint rp, bool isOutgoing, vector<JointEdge> & edges) const;
+  void GetEdgesList(Joint::Id jointId, bool forward, vector<JointEdge> & edges) const;
```

Comment:
You have different names of second param in declaration and in implement of GetEdgesList() method.